.github/workflows/screener-build.yml

```diff
@@ -0,0 +1,172 @@
+      # BUILD_SOURCEBRANCH = branch that triggered the workflow -> refs/pull/<pr_number>/merge
+      ###########################################
+      - name: Set base env variables
+        run: |
+          cat <<EOT >> artifacts/environment
+          BROWSERSLIST_IGNORE_OLD_DATA=true
+          DEPLOYHOST='fluentuipr.z22.web.core.windows.net'
+          BUILD_BUILDID=${{ github.run_id }}
+          BUILD_SOURCEBRANCH=${{ github.ref }}
+          EOT
+
+      ###########################################
+      # SYSTEM_PULLREQUEST_TARGETBRANCH = target branch name
+      # SYSTEM_PULLREQUEST_SOURCEBRANCH = source branch name
+      # SYSTEM_PULLREQUEST_PULLREQUESTID = ID of the PR
+      # SYSTEM_PULLREQUEST_SOURCECOMMITID = commit SHA of PR
+      # ISPR = true if workflow triggered by PR
+      # DEPLOYBASEPATH = path for deploy URL -> pull/<pr_number> for PRs
+      # DEPLOYURL= address for tests deployment, uses DEPLOYHOST and DEPLOYBASEPATH
+      # BUILD_SOURCEBRANCHNAME = 'merge' for PRs
+      ###########################################
+      - name: Set env variables if there is a PR
+        run: |
+          cat <<EOT >> artifacts/environment
+          SYSTEM_PULLREQUEST_TARGETBRANCH=${{ github.event.pull_request.base.ref }}
+          SYSTEM_PULLREQUEST_SOURCEBRANCH=${{ github.event.pull_request.head.ref }}
+          SYSTEM_PULLREQUEST_PULLREQUESTID=${{ github.event.pull_request.id }}
+          SYSTEM_PULLREQUEST_SOURCECOMMITID=${{ github.event.pull_request.head.sha }}
+          ISPR=true
+          DEPLOYBASEPATH=pull/$PR_NUMBER
+          DEPLOYURL=https://$DEPLOYHOST/pull/$PR_NUMBER
+          BUILD_SOURCEBRANCHNAME='merge' ,
+          EOT
+        env:
+          DEPLOYHOST: 'fluentuipr.z22.web.core.windows.net'
+          PR_NUMBER: ${{ github.event.number }}
+        if: ${{startsWith(github.ref, 'refs/pull/')}}
+
+      - name: Set env variables if there is not a PR
+        run: |
+          cat <<EOT >> artifacts/environment
+          ISPR=false
+          DEPLOYBASEPATH=heads/$SYSTEM_PULLREQUEST_TARGETBRANCH
+          DEPLOYURL=https://$DEPLOYHOST/heads/$SYSTEM_PULLREQUEST_TARGETBRANCH
+          BUILD_SOURCEBRANCHNAME=${{ github.ref_name }}
+          EOT
+        if: ${{!startsWith(github.ref, 'refs/pull/')}}
+
+      - name: Upload environment variables artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: env-artifact
+          path: artifacts/environment
+
+  ScreenerNorthStar:
+    runs-on: 'ubuntu-latest'
+
+    name: Screener @fluentui/react-northstar
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.18.1
+
+      - name: Log environment variables (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          printenv | sort ;\
+          echo "SHELLOPTS $SHELLOPTS" ;\
+
+      - name: yarn
+        run: ./yarn-ci.sh
```

Comment:
can we install dependencies only once ? with this workflows it will happen 3x2=6 times. not sure if it will be cached, if yes disregard.

Reply:
Each job is run by a different runner so that the jobs within the same workflow can run in parallel, so dependencies in this case (where jobs do not depend on each other) cannot share dependencies between them.

Reply:
gotcha.

**additional question:** can't we have a job as pre-requirement for all this parallel ones that would install those deps ?

Reply:
There are composite actions [1] that allow re-using the same steps for multiple jobs, they would still need to be used as a step in each job and may require some additional approvals since a composite action is not part of the approved actions from the GitHub marketplace as far as I understand.

[1] https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/